walking:
  engine:
    foot_z_pause: 0.0
    foot_apex_phase: 0.5
    foot_put_down_z_offset: 0.0
    foot_put_down_phase: 1.0
    foot_overshoot_ratio: 0.0
    foot_overshoot_phase: 0.85
    trunk_pause: 0.0
    trunk_x_offset_p_coef_forward: 0.0
    trunk_x_offset_p_coef_turn: 0.0
    first_step_trunk_phase: -0.5

    kick_length: 0.09
    kick_vel: 0.2
    kick_phase: 0.75

    double_support_ratio: 0.286665084270828
    first_step_swing_factor: 1.22922903037799
    foot_distance: 0.239257911972102
    foot_rise: 0.109836395010989
    freq: 1.95742439068112
    trunk_height: 0.33232820113786
    trunk_phase: -0.361252827876859
    trunk_pitch: 0.00603225626629424
    trunk_pitch_p_coef_forward: -1.29662392706207
    trunk_pitch_p_coef_turn: 0.216034705876737
    trunk_swing: 0.567926022073427
    trunk_x_offset: -0.00905033855143576
    trunk_y_offset: 0.00322728505340998
    trunk_z_movement: 0.0241152887755999

  node:
    # update frequency of the engine
    engine_freq: 200.0

    # parameters for bioIK
    ik_timeout: 0.01

    debug_active: True

    trunkYOnlyInDoubleSupport: False

    # 0.5
    # -0.4
    max_step_x: 10
    # 0.12
    max_step_y: 10
    max_step_xy: 10
    max_step_z: 10
    # 0.5
    max_step_angular: 10

    x_speed_multiplier: 1.33
    y_speed_multiplier: 1.5
    yaw_speed_multiplier: 1.5

    vel: -1
    acc: -1
    pwm: -1

    imu_active: False
    imu_pitch_threshold: 0.19
    imu_roll_threshold: 0.4
    imu_pitch_vel_threshold: 1.3
    imu_roll_vel_threshold: 5.7
    pause_duration: 3.0

    phase_reset_active: False
    ground_min_pressure: 1.5
    phase_reset_phase: 0.25
    joint_min_effort: 35
    effort_phase_reset_active: False

  pid_trunk_fused_pitch:
    p: 2.0
    i: 0.05
    d: 0
    i_clamp: 0
    i_clamp_min: -1.0
    i_clamp_max: 1.0
    antiwindup: False
    publish_state: False

  pid_trunk_fused_roll:
    p: 1.0
    i: 0.1
    d: 0
    i_clamp: 0
    i_clamp_min: -1.0
    i_clamp_max: 1.0
    antiwindup: False
    publish_state: False
